### PR TITLE
hw

### PR DESCRIPTION
**Pull request by YaTut1901**

### CODE REVIEW

File: src/main/java/core/basesyntax/Rectangle.java [hunk 1]
```diff
@@ -27,4 +29,30 @@ public Integer getLength() {
...
+        if (this == object) {
+            return true;
+        }
+        if (object == null || getClass() != object.getClass()) {
+            return false;
+        }
+        Rectangle rectangle = (Rectangle) object;
+        return Objects.equals(width, rectangle.width)
+                && Objects.equals(length, rectangle.length)
+                && Objects.equals(color, rectangle.color);
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/Rectangle.java]: please see checklist

File: src/main/java/core/basesyntax/Rectangle.java [hunk 2]
```diff
@@ -27,4 +29,30 @@ public Integer getLength() {
...
+    }
+
+    @Override
+    public int hashCode() {
+        if (width == null || length == null || color == null) {
+            return 0;
+        }
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/Rectangle.java]: if any of your fields is null, you ignore all the others and return null
this isn't right

File: src/main/java/core/basesyntax/Rectangle.java [hunk 3]
```diff
@@ -27,4 +29,30 @@ public Integer getLength() {
...
+        int result = 22;
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/Rectangle.java]: let's use prime numbers

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Rectangle.java (hunk 2 above)
Comment (YaTut1901) [reply src/main/java/core/basesyntax/Rectangle.java]: but it`s required to return int not null

Comment (KseniiaMakarova) [reply src/main/java/core/basesyntax/Rectangle.java]: sorry, I meant 
`if any of your fields is null, you ignore all the others and return 0
this isn't right`

the logic itself is not correct
you should decide whether to change `result` after checking each field, and not them all together

File: src/main/java/core/basesyntax/Rectangle.java [hunk 4]
```diff
@@ -27,4 +27,30 @@ public Integer getLength() {
...
+        if (this == object) {
+            return true;
+        }
+        if (object == null || getClass() != object.getClass()) {
+            return false;
+        }
+        Rectangle rectangle = (Rectangle) object;
+        return width != null && width.equals(rectangle.width) || rectangle.width == width
+                && length != null && length.equals(rectangle.length) || rectangle.length == length
+                && color != null && color.equals(rectangle.color) || rectangle.color == color;
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/Rectangle.java]: The logical-AND operator ( && ) has higher precedence than the logical-OR operator ( || )
add parenthesis to this statement to ensure the flow is correct

File: src/main/java/core/basesyntax/Rectangle.java [hunk 5]
```diff
@@ -27,4 +27,30 @@ public Integer getLength() {
...
+    }
+
+    @Override
+    public int hashCode() {
+        int result = 17;
+        int modifier1 = width == null ? 1 : width;
+        int modifier2 = length == null ? 1 : length;
+        int modifier3 = color == null ? 1 : color.hashCode();
+        result = 31 * result * modifier1;
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/Rectangle.java]:
```suggestion
        result = 31 * result * width == null ? 1 : width.hashCode();
```
please refactor all three lines like this - remove `modifier` variables and use hashCode of the fields

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request closed without merging**